### PR TITLE
check artwork version on order submit

### PR DESCRIPTION
**Pull request by saolsen**
Checks artwork version for line items matches the current version on an order submit. If they differ raises an error.

### CODE REVIEW

**Comment by williardx** [comment]
Thanks @saolsen, looks good 👍 

Do we need to do this for order approval as well?

**Comment by ashkan18** [comment]
@williardx i don't think we need anything on approval, from user's perspective, they should be able to buy the order they have seen and verified. Maybe in CMS UI we want to show something telling partners but i think from Exchange's perspective they should still be allowed to approve it.

File: app/services/order_submit_service.rb [hunk 1]
```diff
@@ -39,6 +39,10 @@ def process!
   private
 
   def pre_process!
+    @order.line_items.map do |li|
+      artwork = GravityService.get_artwork(li[:artwork_id])
+      raise Errors::OrderError, 'Artwork version out of date' if artwork[:current_version_id] != li[:artwork_version_id]
```
Comment (ashkan18) [inline app/services/order_submit_service.rb]: we've removed `Errors::OrderError` in https://github.com/artsy/exchange/pull/172, we need to change this line to
```ruby
raise Errors::ProcessingError, :artwork_version_missmatch if ....
```

**Comment by ashkan18** [comment]
🙌

File: app/services/order_submit_service.rb [hunk 2]
```diff
@@ -39,6 +39,10 @@ def process!
   private
 
   def pre_process!
+    @order.line_items.map do |li|
+      artwork = GravityService.get_artwork(li[:artwork_id])
+      raise Errors::ProcessingError, :artwork_version_missmatch if artwork[:current_version_id] != li[:artwork_version_id]
```
Comment (starsirius) [inline app/services/order_submit_service.rb]: Possible typo "mismatch"?

Comment (ashkan18) [reply app/services/order_submit_service.rb]: 👀 im fixing it in my error handling PR.

Comment (ashkan18) [reply app/services/order_submit_service.rb]: https://github.com/artsy/exchange/pull/184

**Pull request merged**